### PR TITLE
Run license e2e tests

### PR DESCRIPTION
**Pull request by pebrc**
This PR enables the license e2e test on CI. It introduces a Kubernetes secret to the test job and  some changes to the e2e run command to facilitate the mounting of (license) secrets into the e2e test job

### CODE REVIEW

**Comment by pebrc** [comment]
Successful license test run: https://devops-ci.elastic.co/job/cloud-on-k8s-pr/2022/console
Switching back to smoke tests now in PR build

File: Makefile [hunk 1]
```diff
@@ -361,6 +362,7 @@ e2e-run:
 		--operator-image=$(OPERATOR_IMAGE) \
 		--e2e-image=$(E2E_IMG) \
 		--test-regex=$(TESTS_MATCH) \
+		--test-licence=$(TEST_LICENSE) \
```
Comment (pebrc) [inline Makefile]: We need to settle on a spelling variant here :-(

Comment (thbkrkr) [reply Makefile]: What prevents us from fixing it?

File: build/ci/pr/Jenkinsfile [hunk 2]
```diff
@@ -131,7 +131,11 @@ TESTS_MATCH = TestSmoke
 SKIP_DOCKER_COMMAND = false
 IMG_SUFFIX = -ci
 E2E_JSON = true
+GO_TAGS = release
+export LICENSE_PUBKEY = /go/src/github.com/elastic/cloud-on-k8s/build/ci/license.key
+TEST_LICENSE = /go/src/github.com/elastic/cloud-on-k8s/build/ci/test-license.json
 EOF
+       make -C build/ci get-elastic-public-key get-test-license
```
Comment (thbkrkr) [inline build/ci/pr/Jenkinsfile]: Missing bits in the revert?

File: Makefile (hunk 1 above)
Comment (pebrc) [reply Makefile]: Nothing. I think I am going for the US spelling `license` which is consistent with what Elasticsearch has https://www.elastic.co/guide/en/elasticsearch/reference/current/licensing-apis.html

Comment (david-kow) [reply Makefile]: I'd go for license, but that's not a strong opinion.

File: build/ci/pr/Jenkinsfile (hunk 2 above)
Comment (pebrc) [reply build/ci/pr/Jenkinsfile]: Got, it will remove it

**Comment by thbkrkr** [comment]
Almost LGTM.

I think we also need to update `ci/e2e/stack_versions.jenkinsfile`, `ci/e2e/custom_operator_image.jenkinsfile` and `ci/e2e/GKE_k8s_versions.jenkinsfile` with the bits for the license setup.

This shows the need to factorize the code to run E2E test jobs and also to reduce the number of jobs by making them more configurable.

**Comment by pebrc** [comment]
> I think we also need to update ci/e2e/stack_versions.jenkinsfile, ci/e2e/custom_operator_image.jenkinsfile and ci/e2e/GKE_k8s_versions.jenkinsfile with the bits for the license setup.

Well spotted! Fixed (I hope) in f6bbec0

**Pull request merged**